torch_geometric/nn/models/schnet.py

```diff
@@ -224,17 +225,34 @@ def from_qm9_pretrained(root: str, dataset: Dataset, target: int):
 
         return net, (dataset[train_idx], dataset[val_idx], dataset[test_idx])
 
-    def forward(self, z, pos, batch=None):
-        """"""
+    @staticmethod
+    def _validate_fwd_args(z, pos, batch, edge_index, edge_weight):
         assert z.dim() == 1 and z.dtype == torch.long
+
+        if batch is not None:
+            assert batch.dim() == 1 and batch.dtype == torch.long
+
+        if pos is not None:
+            assert edge_index is None and edge_weight is None
```

Comment:
raise maybe a `ValueError`, with a string that explains the error, that will enhance debugability.

Reply:
Decided to remove the details assertions, I'm not sure about this but I was weary of adding a check that might raise exceptions in case that impacts model performance.  

Hopefully the more detailed documentation will help.

Reply:
I think also with your suggestion to drop providing the `edge_weight` argument it is much simpler and easier to use the model correctly.